### PR TITLE
feat: Added mail settings attributes for bypass_spam_management, bypass_bou…

### PR DESCRIPTION
**Pull request by asos-vinodpatil**
…nce_management and bypass_unsubscribe_management

https://github.com/sendgrid/sendgrid-csharp/issues/1135

# Fixes #

A short description of what this PR does.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/sendgrid/sendgrid-csharp/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://support.sendgrid.com), or create a GitHub Issue in this repository.

### CODE REVIEW

**Review by JenniferMah — COMMENTED** [review]
Hi @asos-vinodpatil! Thanks for your PR! Would you mind adding some documentation for spam, bounce, and unsubscribe bypass management filters in the [Usage.md](https://github.com/sendgrid/sendgrid-csharp/blob/main/USAGE.md) and [Use_cases.md](https://github.com/sendgrid/sendgrid-csharp/blob/main/USE_CASES.md)? Here's an example of the [documentation in php](https://github.com/sendgrid/sendgrid-php/pull/1065/files?short_path=58a825c#diff-58a825c70bccfbe691d7ddbddb133275f4f657de6aeb4a502fa9fce279f8bf8d).

**Comment by asos-vinodpatil** [comment]
> Hi @asos-vinodpatil! Thanks for your PR! Would you mind adding some documentation for spam, bounce, and unsubscribe bypass management filters in the [Usage.md](https://github.com/sendgrid/sendgrid-csharp/blob/main/USAGE.md) and [Use_cases.md](https://github.com/sendgrid/sendgrid-csharp/blob/main/USE_CASES.md)? Here's an example of the [documentation in php](https://github.com/sendgrid/sendgrid-php/pull/1065/files?short_path=58a825c#diff-58a825c70bccfbe691d7ddbddb133275f4f657de6aeb4a502fa9fce279f8bf8d).

Thanks @JenniferMah, added required documentation.

**Comment by asos-vinodpatil** [comment]
@JenniferMah Any update on this PR

File: src/SendGrid/Helpers/Mail/Model/BypassBounceManagement.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// Copyright (c) Twilio SendGrid. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// </copyright>
+
+using Newtonsoft.Json;
+
+namespace SendGrid.Helpers.Mail
+{
+    /// <summary>
+    /// Allows you to bypass all unsubscribe groups and suppressions to ensure that the email is delivered to every single recipient. This should only be used in emergencies when it is absolutely necessary that every recipient receives your email. Ex: outage emails, or forgot password emails.
```
Comment (shwetha-manvinkurke) [inline src/SendGrid/Helpers/Mail/Model/BypassBounceManagement.cs:11]: update with the right doc

File: src/SendGrid/Helpers/Mail/Model/BypassSpamManagement.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// Copyright (c) Twilio SendGrid. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// </copyright>
+
+using Newtonsoft.Json;
+
+namespace SendGrid.Helpers.Mail
+{
+    /// <summary>
+    /// Allows you to bypass all unsubscribe groups and suppressions to ensure that the email is delivered to every single recipient. This should only be used in emergencies when it is absolutely necessary that every recipient receives your email. Ex: outage emails, or forgot password emails.
```
Comment (shwetha-manvinkurke) [inline src/SendGrid/Helpers/Mail/Model/BypassSpamManagement.cs:11]: update with the right doc

File: src/SendGrid/Helpers/Mail/Model/BypassBounceManagement.cs (hunk 1 above)
Comment (asos-vinodpatil) [reply src/SendGrid/Helpers/Mail/Model/BypassBounceManagement.cs]: Done

File: src/SendGrid/Helpers/Mail/Model/BypassSpamManagement.cs (hunk 2 above)
Comment (asos-vinodpatil) [reply src/SendGrid/Helpers/Mail/Model/BypassSpamManagement.cs]: Done

**Comment by rakatyal** [comment]
Thank you for the PR. LGTM!

**Review by rakatyal — APPROVED** [review]

**Pull request merged**